### ISSUE TITLE
Difference in paper and code

### ISSUE THREAD

**Issue by anatolix** [ZheC/Realtime_Multi-Person_Pose_Estimation#118]
**Paper says:**
<img width="511" alt="screen shot 2017-11-14 at 00 05 40" src="https://user-images.githubusercontent.com/1033375/32752912-04cc843a-c8dc-11e7-935f-9f15dabcc943.png">

**Code is following**
https://github.com/CMU-Perceptual-Computing-Lab/caffe_train/blob/master/src/caffe/cpm_data_transformer.cpp
<img width="1077" alt="screen shot 2017-11-14 at 01 33 49" src="https://user-images.githubusercontent.com/1033375/32752950-1cfcf1ac-c8dc-11e7-9a03-a2e089af86bb.png">

Why is that?

**Comment by ZheC** [comment]
Yes, this is actually a bug. It should be min(max(...), 1.0). Thank you for pointing out.

**Comment by anatolix** [comment]
@ZheC thank you.

p.s. Speaking about a bugs - probably you have another one, some PAF could became NaN, I've pointed to the line of code here 
https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/issues/102

please check it too.

**Comment by last-one** [comment]
Another difference about PAF.
paper says:
![screen-2017-11-16-20-00-06](https://user-images.githubusercontent.com/9068170/32890273-053ac8a8-cb09-11e7-8453-457e6d7caf60.jpg)
the code is:
![screen-2017-11-16-20-03-05](https://user-images.githubusercontent.com/9068170/32890318-3cc7e350-cb09-11e7-99db-b4fa269f6108.jpg)

If don't update the value of cnt when cnt = 0. The value of cnt always will be 0. And the value of paf will be the last part value, not the average of the affinity fields of all people in the image.

So I think ther is no need to make a special judgment when cnt = 0.

**Comment by Hanhanhan11** [comment]
> **Paper says:**
> <img alt="screen shot 2017-11-14 at 00 05 40" width="511" src="https://user-images.githubusercontent.com/1033375/32752912-04cc843a-c8dc-11e7-935f-9f15dabcc943.png">
> 
> **Code is following**
> https://github.com/CMU-Perceptual-Computing-Lab/caffe_train/blob/master/src/caffe/cpm_data_transformer.cpp
> <img alt="screen shot 2017-11-14 at 01 33 49" width="1077" src="https://user-images.githubusercontent.com/1033375/32752950-1cfcf1ac-c8dc-11e7-9a03-a2e089af86bb.png">
> 
> I.e. not max, but min(sum(...),1.0)

I can not find which line, the picture you show did not contain max op.
